From: Suren Baghdasaryan <surenb@google.com>
Subject: alloc_tag: export memory allocation profiling symbols used by modules
Date: Thu, 11 Jul 2024 13:46:26 -0700

Export mem_alloc_profiling_key, page_ext_get() and page_ext_set() symbols
as they can be used by modules (mem_alloc_profiling_key is used indirectly
via mem_alloc_profiling_enabled()).

Link: https://lkml.kernel.org/r/20240711204626.1669138-1-surenb@google.com
Fixes: 22d407b164ff ("lib: add allocation tagging support for memory allocation profiling")
Signed-off-by: Suren Baghdasaryan <surenb@google.com>
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202407080044.DWMC9N9I-lkp@intel.com/
Cc: Kees Cook <keescook@chromium.org>
Cc: Kent Overstreet <kent.overstreet@linux.dev>
Cc: Pasha Tatashin <pasha.tatashin@soleen.com>
Cc: Sourav Panda <souravpanda@google.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/alloc_tag.c |    1 +
 mm/page_ext.c   |    2 ++
 2 files changed, 3 insertions(+)

--- a/lib/alloc_tag.c~alloc_tag-export-memory-allocation-profiling-symbols-used-by-modules
+++ a/lib/alloc_tag.c
@@ -15,6 +15,7 @@ EXPORT_SYMBOL(_shared_alloc_tag);
 
 DEFINE_STATIC_KEY_MAYBE(CONFIG_MEM_ALLOC_PROFILING_ENABLED_BY_DEFAULT,
 			mem_alloc_profiling_key);
+EXPORT_SYMBOL(mem_alloc_profiling_key);
 
 struct allocinfo_private {
 	struct codetag_iterator iter;
--- a/mm/page_ext.c~alloc_tag-export-memory-allocation-profiling-symbols-used-by-modules
+++ a/mm/page_ext.c
@@ -527,6 +527,7 @@ struct page_ext *page_ext_get(const stru
 
 	return page_ext;
 }
+EXPORT_SYMBOL(page_ext_get);
 
 /**
  * page_ext_put() - Working with page extended information is done.
@@ -545,3 +546,4 @@ void page_ext_put(struct page_ext *page_
 
 	rcu_read_unlock();
 }
+EXPORT_SYMBOL(page_ext_put);
_
